src/content/posts/2020-10-08-custom-kerberos-integration-using-the-mysql-protocol-in-the-browser.md

```diff
@@ -0,0 +1,64 @@
+---
+template: blog-post
+title: Custom Kerberos integration using the MySQL protocol in the browser
+slug: kerberos-mysql-browser
+date: 2020-10-08 14:06
+description: Custom Kerberos integration using the MySQL protocol in the browser
+---
+When you put these 3 words together: Kerberos, MySQL authentication, and a Browser, they can look quite daunting at first glance.
+
+My goal with this post is to briefly explain how we integrated them all in a very simple and neat way.
+
+I am not going too deep in the concepts I just mentioned but mainly will focus on the solution itself. There are plenty of resources already across the whole internet explaining those concepts.
+
+
+**Use Case**
+
+The application in the matter is memSQL studio, a stateless UI to manage and interact with a memSQL cluster.
+
+First of all, let’s take a quick look at its architecture:
+
+* Studio has no state. The UI is simply a tool to interact with a cluster in a simpler way. All the fetched/shown data and authentication logic is delegated to the cluster.
+* MemSQL wire protocol is MySQL compatible, which allows us to use any existing MySQL driver.
+* Studio can run without any backend server and solely inside a browser.
+* For communication with the cluster we use a forked version of the mysqljs driver inside a browser worker thread. We can say that the Studio “backend” lives inside the worker.
```

Comment:
If you want, you can link to this blog post which explains the architecture: https://www.memsql.com/blog/web-workers-client-side-react-redux/